src/jltypes.c

```diff
@@ -3762,8 +3762,13 @@ void jl_init_types(void) JL_GC_DISABLED
                                         "_state",
                                         "sticky",
                                         "_isexception",
-                                        "priority"),
-                        jl_svec(16,
+                                        "priority",
+                                        "metrics_enabled",
+                                        "first_enqueued_at",
+                                        "last_started_running_at",
+                                        "cpu_time_ns",
+                                        "finished_at"),
```

Comment:
Would it make sense to instead store this state out-of-line and just have a single pointer here?

You could be cute and have a Union{Nothing,Metric} for the metric enabled query.
Alternativly these values could live in task-local-storage.

Reply:
We originally proposed that, yeah, but they were moved inline. I don't remember why. Maybe to dodge the extra cache-miss on the task switches?

CC @vtjnash